### DISCUSSION TITLE
Full Screen Window

### DISCUSSION THREAD

**Discussion by Farbodsol** [TomSchimansky/CustomTkinter#453] [Q&A]
Hi,

Thanks for the wonderful library!

I'm attempting to make the main window a full screen when the app first runs.  I created a function to place the window at the center and as a full screen. Part of the code is as follows:
self.WIDTH= self.winfo_screenwidth()
self.HEIGHT= self.winfo_screenheight()
self.geometry(f'{self.WIDTH}x{self.HEIGHT}+0+0')

The issue appears when the main window is instantiated and runs on full screen with **gaps horizontally and vertically to the top left corner of the monitor**. I'd like to draw your attention to the point that when I call the normal root=tk.TK() the window appears as full screen and in the right place. No matter what value is assigned to the positioning terms of self. geometry, the window pops up with either a slight shift to right or left.
Slight right of right corner: self.geometry(f'{self.WIDTH}x{self.HEIGHT}+0+0')
Slight left of monitor left corner: self.geometry(f'{self.WIDTH}x{self.HEIGHT}-0+0')

The center function that I normally use and adopted from another user is as follows:

self.WIDTH= self.winfo_screenwidth()
self.HEIGHT= self.winfo_screenheight()
self.geometry(f'{self.WIDTH}x{self.HEIGHT}+0+0')
self.center() 

def center(self):
    """
    centers a tkinter window
    :param win: the main window or Toplevel window to center
    """
    win.update_idletasks()
    width = win.winfo_width()
    frm_width = win.winfo_rootx() - win.winfo_x()
    win_width = width + 2 * frm_width
    height = win.winfo_height()
    titlebar_height = win.winfo_rooty() - win.winfo_y()
    win_height = height + titlebar_height + frm_width
    x = win.winfo_screenwidth() // 2 - win_width // 2
    y = win.winfo_screenheight() // 2 - win_height // 2
    win.geometry('{}x{}+{}+{}'.format(width, height, x, y))
    win.deiconify()
 

I'd appreciate it if you assist me to tackle this issue. I guess the solution might be quite simple, however, I'm new to python scripting and this library.

Thanks.

**Comment by OleksandrMotsok** [comment]
Hi, @Farbodsol,

I am also working on a full-screen window. Maybe my approach will help you.
I am doing 
app = customtkinter.CTk()  
app.attributes("-fullscreen", True)
And after it the application runs in full-screen mode.

Have a nice day

**Reply by Farbodsol** [reply]
Hi @OleksandrMotsok,
Thanks for your response! The issue with setting the full screen attribute to true is that the top bar with close, minimize/maximize buttons disappears, which is not what exactly I like to have. If you run the “center” function from my post with regular tk.Tk() main window, you’ll notice that the window runs in full screen with all other buttons active. However, using customtkinter.CTk() a small gap is left between the top left corner of the monitor and created app.
Best,

**Comment by Hancie123** [comment]
try using
self.root.state('zoomed')
